src/cmd/services/m3query/config/config.go

```diff
@@ -150,10 +150,33 @@ type CarbonConfiguration struct {
 
 // CarbonIngestionConfiguration is the configuration struct for carbon ingestion.
 type CarbonIngestionConfiguration struct {
-	Enabled        *bool          `yaml:"enabled"`
-	MaxConcurrency int            `yaml:"maxConcurrency"`
-	ListenAddress  string         `yaml:"listenAddress"`
-	Timeout        *time.Duration `yaml:"timeout"`
+	Enabled        *bool                              `yaml:"enabled"`
+	MaxConcurrency int                                `yaml:"maxConcurrency"`
+	ListenAddress  string                             `yaml:"listenAddress"`
+	Timeout        *time.Duration                     `yaml:"timeout"`
+	Rules          []CarbonIngestionRuleConfiguration `yaml:"rules"`
+}
+
+// CarbonIngestionRuleConfiguration is the configuration struct for a carbon
+// ingestion rule.
+type CarbonIngestionRuleConfiguration struct {
+	Pattern  string                                      `yaml:"pattern"`
+	Policies []CarbonIngestionStoragePolicyConfiguration `yaml:"policies"`
+}
+
+// CarbonIngestionStoragePolicyConfiguration is the configuration struct for
+// a carbon rule's storage policies.
+type CarbonIngestionStoragePolicyConfiguration struct {
```

Comment:
TODO: Write getters to make these all have reasonable defaults as we discussed @robskillington